src/hooks/cgroups/pbs_cgroups.PY

```diff
@@ -2578,6 +2578,27 @@ class CgroupUtils(object):
                 pbs.logmsg(pbs.EVENT_DEBUG4, "Insufficient ncpus: %s/%s" %
                            (req, available['cpus']))
                 return {}
+
+        if 'mem' in requested or 'ngpus' in requested or 'nmics' in requested:
+            # for multivnode systems asking memory/ngpus/nmics without asking cpus is valid, 
+            # need access to memory
+            if 'cpuset.mems' not in assigned:
+                assigned['cpuset.mems'] = socketlist
+            if 'cpuset.cpus' not in assigned:
+                assigned['cpuset.cpus'] = []
+
+        if 'mem' not in requested and 'ncpus' not in requested \
+          and ('vmem' in requested or 'ngpus' in requested or 'nmics' in requested):
+            # return something to mark that some of the swap or devices nominally assigned 
+            # to this vnode is indeed being assigned; dummy entry to avoid error because 
+            # "assignment failed"
+            # hopefully there will be something else on the host requiring cpus or memory and
+            # that will be merged with this
+            if 'cpuset.cpus' not in assigned:
+                assigned['cpuset.cpus'] = []
+            if 'cpuset.mems' not in assigned:
+                assigned['cpuset.mems'] = []
```

Comment:
Can you simplify this logic by just having the above 4 lines outside of the two main ifs?  You can then have the first if for setting cpuset.mems to socketlist.  It could mean that you are returning assigned with an empty list in other cases.  Is that bad?

Reply:
I moved them to the very top in `assigned` definition.